uni/lib/view/common_widgets/pages_layouts/general/general.dart

```diff
@@ -34,8 +37,15 @@ abstract class GeneralPageViewState<T extends StatefulWidget> extends State<T> {
       try {
         await onLoad(context);
       } catch (e, stackTrace) {
-        Logger().e('Failed to load page info: $e\n$stackTrace');
-        await Sentry.captureException(e, stackTrace: stackTrace);
+        final connectivity = await Connectivity().checkConnectivity();
+        if (connectivity == ConnectivityResult.none) {
+          setState(() {
+            _connected = false;
+          });
```

Comment:
Connectivity is not a reliable way to detect if the user can connect to the page. For instance, the user can be connected to a wifi network and still not be able to get the page (the wifi itself may not have connectivity).